MVCapplication/src/main/webapp/WEB-INF/resources/js/emptyCart.js

```diff
@@ -0,0 +1,17 @@
+function waitForElement(){
+    if(typeof connected !== 'undefined' && connected === true){
+        if (document.referrer.match(/.*\/cart([\/?].*)?/g)) {
+            $.ajax({
+                url: "/shopping/finished",
+                method: "POST"
+            });
+        }
+    }
+    else{
+        setTimeout(waitForElement, 250);
+    }
+}
+$(document).ready(function () {
+    waitForElement();
```

Comment:
this function waits for establishing WebSocket connection, and then make request to print message about finishing shopping